bundles/org.openhab.binding.tacmi/src/main/java/org/openhab/binding/tacmi/internal/json/TACmiJsonHandler.java

```diff
@@ -0,0 +1,217 @@
+        }
+        // we want to trigger the initial refresh 'at once'
+        this.scheduledFuture = scheduler.scheduleWithFixedDelay(this::refreshData, 0, pollInterval, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // We do not support commands
+    }
+
+    @Override
+    public void dispose() {
+        final ScheduledFuture<?> scheduledFuture = this.scheduledFuture;
+        if (scheduledFuture != null) {
+            scheduledFuture.cancel(true);
+            this.scheduledFuture = null;
+        }
+        super.dispose();
+    }
+
+    private void refreshData() {
+        try {
+            final Request req = httpClient.newRequest(this.url).method(HttpMethod.GET).timeout(10000,
+                    TimeUnit.MILLISECONDS);
+            req.header(HttpHeader.ACCEPT_LANGUAGE, "en");
+            final String ah = this.authHeader;
+            if (ah != null) {
+                req.header(HttpHeader.AUTHORIZATION, ah);
+            }
+
+            final ContentResponse res = req.send();
+            if (res.getStatus() != 200) {
+                logger.warn("Error requesting update {} / {} \n{}", res.getStatus(), res.getReason(),
+                        res.getContentAsString());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                return;
+            }
+            logger.trace("Reply:\n{}", res.getContentAsString());
+            Gson gson = new GsonBuilder().setFieldNamingPolicy(FieldNamingPolicy.UPPER_CAMEL_CASE).create();
+            JsonResponse resp = gson.fromJson(res.getContentAsString(), JsonResponse.class);
+            if (resp == null) {
+                logger.error("Response is Empty");
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                return;
+            }
+            if (resp.statusCode != 0) {
+                logger.error("Response Error: {} ({})", resp.statusCode, resp.status);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, resp.status);
+                return;
+            }
+            updateChannels(resp);
+            updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE);
+        } catch (final InterruptedException e) {
+            // binding shutdown is in progress
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE);
+        } catch (final Exception e) {
+            logger.trace("Exception:\n{}", e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Error: " + e.getMessage());
+        }
+    }
+
+    private void updateChannels(JsonResponse jr) throws Exception {
+        final var allchans = new ArrayList<Channel>();
+        boolean mod = false;
+
+        BiFunction<Collection<IO>, String, Boolean> chanhandler = (coll, pre) -> {
+            var lmod = false;
+            for (var inp : coll) {
```

Comment:
General comment that applies to the whoel file: Many abbreviations are used for variables, please use meaningfull names. This would make it easier to troubleshoot, review and maintain the code.